freeipa/src/main/java/com/sequenceiq/freeipa/service/AltusMachineUserService.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.freeipa.service;
+
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.model.AltusCredential;
+import com.sequenceiq.cloudbreak.auth.altus.service.AltusIAMService;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.util.CrnService;
+
+@Service
+public class AltusMachineUserService {
+
+    private static final String FREEIPA_FLUENT_DATABUS_MACHINE_USER_PATTERN = "freeipa-fluent-databus-uploader-%s";
+
+    private final CrnService crnService;
+
+    private final AltusIAMService altusIAMService;
+
+    public AltusMachineUserService(AltusIAMService altusIAMService, CrnService crnService) {
+        this.altusIAMService = altusIAMService;
+        this.crnService = crnService;
+    }
+
+    public Optional<AltusCredential> createMachineUserWithAccessKeys(Stack stack) {
+        String userCrn = crnService.getUserCrn();
+        String machineUserName = getFluentMachineUser(stack);
+        return altusIAMService.generateMachineUserWithAccessKey(machineUserName, userCrn);
```

Comment:
How this `generateMachineUserWithAccessKey` will behave if we restart the service during provision and it is called again?

Reply:
it will generate a new access key (let's say the machine user already exists because of the re-run)..reason is for that we can only get the private key secret during creation. but on termination it deletes all the access keys